solutions/cses-1625.mdx

```diff
@@ -0,0 +1,349 @@
+---
+id: cses-1625
+source: CSES (TODO -- convert to something like `USACO Silver 2017 January`)
+title: Grid Paths
+author: Vivian Han
+---
+
+## Explanation
+
+Adapted from https://usaco.guide/CPH.pdf#page=61
+
+<Warning title="Run times">
+Please note: the running times and number of recursive calls listed are for a separate but very similar problem (# paths from upper left to *lower right* corner on a 7x7 grid). 
+They may not be entirely accurate, but they illustrate the effect of each optimization. 
+The data is also C++ specific, so Java and Python users will experience significantly different runtimes.
+The consequences of this discrepancy which will be addressed later in the explanation.
+</Warning>
+
+### Basic Algorithm
+
+The first version of the algorithm does not contain any optimizations. We simply
+use backtracking to generate all possible paths from the upper-left corner to
+the lower-right corner and count the number of such paths.
+
+• running time: 483 seconds
+
+• number of recursive calls: 76 billion
+
+### Optimization 1
```

Comment:
I think it would be better to wrap these "Optimization ..." in spoilers.

Reply:
https://usaco.guide/general/working-mdx?lang=cpp#spoilers

Reply:
They are labeled as "Optimizations," but they are essential to the solution. To me, it would seem like putting every paragraph of any other editorial in spoilers.

Reply:
sure